server/pulp/plugins/util/misc.py

```diff
@@ -142,3 +143,89 @@ def clear_directory(path, skip_list=()):
 
         elif os.path.isfile(entry_path):
             os.unlink(entry_path)
+
+
+def common_path(dirs):
+    _common_path = os.path.commonprefix(dirs)
+    if not _common_path.endswith("/"):
+        _common_path = os.path.dirname(_common_path)
+    return _common_path
+
+
+def make_link_unit(unit, extra_dst_path, working_dir,
+                   remote_repo_path, remote_root, configuration, extra_src_path, out_path):
+    """
+    This method creates symlink in working directory, pointing to remote
+    content directory, so they can be sync to remote server
+
+    :param repo_type: type of repository
+    :type repo_type: str
+    :param unit: Pulp content unit
+    :type unit: Unit
+    :param extra_dst_path: ending path for processes units
+                          (for example 'Packages' for rpm)
+    :type extra_dst_path: str
+    :param working_dir: Working directory
+    :type working_dir: str
+    :param remote_repo_path: relative repo destination path on remote server
+    :type remote_repo_path: str
+    :param remote_root: remote destination root directory
+    :type remote_root: str
+    :param configuration: configuration of distributor
+    :type configuration: dict
+    """
+    filename = os.path.basename(unit.storage_path)
+
+    if not os.path.exists(os.path.join(working_dir, extra_dst_path, *extra_src_path)):
+        os.makedirs(os.path.join(working_dir, extra_dst_path, *extra_src_path))
+
+    origin_path = get_origin_rel_path(unit, configuration)
+
+    dest = os.path.normpath(os.path.join(working_dir,
+                                         *([extra_dst_path] + extra_src_path + [filename])))
+    link_source = os.path.relpath(os.path.normpath(os.path.join(remote_root,
+                                                                os.path.dirname(origin_path))),
+                                  os.path.normpath(os.path.join(remote_root,
+                                                                remote_repo_path.lstrip("/"),
```

Comment:
As with somewhere else, if there is a leading "/", it's not a relative path. It should fail validation as part of user input, and here it should raise a ``ValueError`` or similar rather than trying to "fix" it.